src/tensor_parallel/slicing_configs.py

```diff
@@ -85,6 +85,46 @@ def split_alibi(alibi: torch.Tensor, *, rank: int, num_heads: int, world_size: i
     return alibi_part.reshape(-1, *alibi.shape[1:])
 
 
+def get_t5_config(model_config: T5Config, devices: Sequence[torch.device]) -> Config:
+    world_size = len(devices)
+    num_heads = model_config.num_heads
+    head_dim = model_config.d_kv
+    gather_kv_across_ranks = CollectiveOperation(
```

Comment:
please make sure you use this op

Reply:
I've rewritten this part a bit. Now those gathers are more or less universal